tests/cpp/test_sdpa_node.cpp

```diff
@@ -82,6 +69,116 @@ auto validateSdpaFwdOutputs = [](std::vector<at::Tensor> nvf_out,
   EXPECT_TRUE(at::equal(nvf_out[6], debug_attn_mask));
 };
 
+void checkSdpaFwdMapping(Fusion* fusion, Expr* op) {
+  IdModel id_model(fusion);
+  const ValGraph& vg = id_model.idGraph(IdMappingMode::EXACT);
+  vg.validateConsistency();
+
+  auto sdpa_op = dynamic_cast<SdpaFwdOp*>(op);
+  ASSERT_TRUE(sdpa_op != nullptr);
+
+  /* SdpaFwdOp:
```

Comment:
Are we checking the S domains of key and value producers are mapped? Also, the E domains of query and key producers?

Reply:
There is no consumer for those, and we don't map the producers in ID model for these two ops.

Reply:
Should we map them in the ID model then?

Reply:
Yes, let's map them together.

Reply:
There are two ways:
1. Adding reduction axis for these domains in the output (similar to having K as a reduction axis in matmul, linear, output)
2. Mapping the producers explicitly in `id_model.cpp::buildExactGraph`.

The former avoids special cases in ID model code but adds the complexity of 2 additional reduction axis.

Reply:
Are they reduction domains? If so, #1 seems to make most sense.

Reply:
@naoyam: Are there any cons to leaving these two domains unmapped?
We ran into an issue when the reduction dimension is 1 in matmul (https://github.com/NVIDIA/Fuser/issues/2532). In this case, we are considering removing the reduction axis from the output.

A reduction axis in the consumer in matmul is required for translating to `MmaOp`. Since SDPA operators would not be nvFuser scheduled, these are not functionally required.

Wdyt?

Reply:
If it's not trivial, don't bother for now. I thought it'd be just adding some things and add additional mappings, which may be useful in some cases. Faithfully modeling domain relationships is always a good thing, but given that there's no known use case, it's fine to leave as a future to-do item. Please make sure to leave an appropriate comment.

Reply:
I'll go ahead with leaving these unmapped to avoid issues similar to (#2532) with SDPA. Will revisit if any use case comes up after integrating with Thunder.